SYCL/Regression/stream_exception_no_alloc.cpp

```diff
@@ -0,0 +1,30 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: env SYCL_PI_TRACE=2 %CPU_RUN_PLACEHOLDER %t.out 2>&1 %CPU_CHECK_PLACEHOLDER
+// RUN: env SYCL_PI_TRACE=2 %GPU_RUN_PLACEHOLDER %t.out 2>&1 %GPU_CHECK_PLACEHOLDER
+// RUN: env SYCL_PI_TRACE=2 %ACC_RUN_PLACEHOLDER %t.out 2>&1 %ACC_CHECK_PLACEHOLDER
+
+// NOTE: This test checks that sycl::stream does not allocate memory if it fails
+// during construction.
+
+#include <CL/sycl.hpp>
+
+#include <limits>
+
+int main() {
+  sycl::queue q;
+  q.submit([&](sycl::handler &cgh) {
+    try {
+      // Try to create stream with invalid workItemBufferSize parameter.
```

Comment:
Can we have a test in `intel/llvm` `sycl/unittests` instead?

Reply:
I was considering it, but it was simpler to have `single_task` try and manifest the memory.

One way to do it in the unittests would be to check that `StreamBufferPool` remains empty, but since it it private to the scheduler I would need to use the mock scheduler. Do you reckon that would work?

Reply:
You can have the same code + PIImages mocks + redefine call of piMemBufferCreate to a function which sets global flag if it is called. Then is "main" check that the flag is not set. Will it work?

Reply:
You are absolutely right, that works. I have moved this to a unittest in https://github.com/intel/llvm/pull/4594 and will close this.